chain/chain/src/block_processing_utils.rs

```diff
@@ -43,6 +45,19 @@ impl From<AddError> for near_chain_primitives::Error {
     }
 }
 
+/// Results from processing a block that are useful for client and client actor to use
+/// for steps after a block is processed that can't be finished inside Chain after a block is processed
+/// (for example, sending requests for missing chunks or challenges).
+/// This struct is passed to Chain::process_block as an argument instead of returned as Result,
+/// because the information stored here need to returned whether process_block succeeds or returns an error.
+#[derive(Default)]
+pub struct BlockProcessingArtifact {
```

Comment:
This name is a bit of a mouthful... Can we do shorter? Perhaps something like `BlockEffects`?

Reply:
Will refactor that later. Renaming this will cause a lot of merge conflict with another PR that I have right now.